docs/cookbook/images-gridfs.rst

```diff
@@ -0,0 +1,645 @@
+
+.. code-block:: php
+
+    namespace AppBundle\Document\Product\Image;
+
+    use AppBundle\Document;
+    use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+
+    /**
+     * @ODM\Document
+     */
+    final class Cache extends Document\Product\Image
+    {
+        /**
+         * @ODM\EmbedOne(
+         *     targetDocument=Cache\Metadata::class,
+         * )
+         *
+         * @var Cache\Metadata
+         */
+        private $metadata;
+
+        /**
+         * @param string $file
+         * @param string $contentType
+         * @param string $filename
+         * @param string $filter
+         */
+        public function __construct($file, $contentType, $filename, $filter)
+        {
+            $this
+                ->setContentType($contentType)
+                ->setFile($file)
+                ->setFilename($filename);
+
+            $this->metadata = new Cache\Metadata($filter);
+        }
+
+        /**
+         * @return Cache\Metadata
+         */
+        public function getMetadata()
+        {
+            return $this->metadata;
+        }
+    }
+
+.. code-block:: php
+
+    namespace AppBundle\Document\Product\Image\Cache;
+
+    use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+
+    /**
+     * @ODM\EmbeddedDocument(
+     *     indexes={
+     *         @ODM\Index(keys={"filter"="asc"}),
+     *     }
+     * )
+     */
+    final class Metadata
+    {
+        /**
+         * @ODM\Field(type="string")
+         *
+         * @var string
+         */
+        private $filter;
+
+        /**
+         * @param string $filter
+         */
+        public function __construct($filter)
+        {
+            $this->filter = $filter;
+        }
+
+        /**
+         * @return string
+         */
+        public function getFilter()
+        {
+            return $this->filter;
+        }
+    }
+
+How to store images in MongoDB GridFS?
+--------------------------------------
+
+In Sylius the `KnpGaufretteBundle <https://github.com/KnpLabs/KnpGaufretteBundle>`_ is used to store images.
+In order to store images in MongoDB GridFS, we have to create new GridFS loader services for Gaufrette:
+
+First of all a new service is introduced in ``@AppBundle/Resources/config/services.yml``.
+
+.. code-block:: yml
+
+    services:
+      app.gaufrette_loader.doctrine_grid_fs:
+        class: Doctrine\MongoDB\GridFS
+        factory: ["@doctrine.odm.mongodb.document_manager", "getDocumentCollection"]
+        arguments: ["AppBundle\\Document\\Product\\Image"]
+        public: false
+
+      app.gaufrette_loader.grid_fs:
+        class: MongoGridFS
+        factory: ["@app.gaufrette_loader.doctrine_grid_fs", "getMongoCollection"]
+
+Now we can override the Gaufrette configuration in ``app/config/config.yml`` to use the newly created
+loader service ``app.gaufrette_loader.grid_fs``.
+
+.. code-block:: yml
+
+    knp_gaufrette:
+      adapters:
```

Comment:
The indentation of yaml files in Sylius is 4 spaces instead of 2 :) Also in the docs.